src/AWS.Deploy.ServerMode.Client/CommandLineWrapper.cs

```diff
@@ -43,9 +49,28 @@ public virtual async Task<int> Run(string command, params string[] stdIn)
                 await process.StandardInput.WriteLineAsync(line).ConfigureAwait(false);
             }
 
+            process.OutputDataReceived += (sender, e) =>
+            {
+                strOutput.AppendLine(e.Data);
+            };
+
+            process.ErrorDataReceived += (sender, e) =>
+            {
+                strError.AppendLine(e.Data);
+            };
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+
             process.WaitForExit(-1);
 
-            return await Task.FromResult(process.ExitCode).ConfigureAwait(false);
+            var result = new RunResult
+            {
+                ExitCode = process.ExitCode,
+                StandardError = strError.ToString(),
```

Comment:
If the deploy tool is running for a long time and if we aren't careful what we write to stdout or stderr this string builders could grow very long. We should probably cap what we store so we don't have an indefinitely growing string StringBuilder behind the scenes. Also we should probably only return back to VS the last few lines.

Reply:
I created a CappedStringBuilder which essentially limits the number of lines the StringBuilder can hold.